### PR TITLE
WorkItemFiler accepts and validates a SARIF log file

### PR DESCRIPTION
**Pull request by lgolding**

### CODE REVIEW

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

**Comment by lgolding** [comment]
@harleenkohli is added to the review. #Closed

File: src/Sarif.WorkItemFiling/WorkItemFiler.cs [hunk 1]
```diff
@@ -1,9 +1,100 @@
...
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        }
+
+        /// <summary>
+        /// Files work items from the results in a SARIF log file.
+        /// </summary>
+        /// <param name="path">
+        /// The path to the SARIF log file.
+        /// </param>
+        public void FileWorkItems(string path)
```
Comment (harleenkohli) [inline src/Sarif.WorkItemFiling/WorkItemFiler.cs]: >path [](start = 41, length = 4)

could we name this something like logFilePath for quicker context?

File: src/Sarif.WorkItemFiling/WorkItemFiler.cs [hunk 2]
```diff
@@ -1,9 +1,100 @@
...
+        {
+            if (path == null) { throw new ArgumentNullException(nameof(path)); }
+
+            ValidateLogFileAgainstSarifSchema(path);
```
Comment (harleenkohli) [inline src/Sarif.WorkItemFiling/WorkItemFiler.cs]: >ValidateLogFileAgainstSarifSchema [](start = 12, length = 33)

suggestion: what about a name like "EnsureIsValidSarif" etc?

File: src/Sarif.WorkItemFiling/WorkItemFiler.cs (hunk 1 above)
Comment (lgolding) [reply src/Sarif.WorkItemFiling/WorkItemFiler.cs]: Good rename. Will do in next PR.

File: src/Sarif.WorkItemFiling/WorkItemFiler.cs (hunk 2 above)
Comment (lgolding) [reply src/Sarif.WorkItemFiling/WorkItemFiler.cs]: Good rename. Will do in next PR.

**Pull request merged**